demo/data/package.json

```diff
@@ -0,0 +1,52 @@
+{
+  "name": "@raincatcher/demo-data",
```

Comment:
I think it's totally okay if it helps in that front. For the local dev workflow and lerna nothing would change, only for npm. (and arguably the `node_modules/@raincatcher` + `node_modules/@raincatcher-demo` directory structure, which shouldn't matter)